qiita_pet/handlers/analysis_handlers/tests/test_sharing_handlers.py

```diff
@@ -0,0 +1,59 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import main
+from json import loads
+
+from qiita_db.analysis import Analysis
+from qiita_db.user import User
+from qiita_pet.test.tornado_test_base import TestHandlerBase
+
+
+class TestShareStudyAjax(TestHandlerBase):
+
+    def test_get(self):
+        a = Analysis(1)
+        u = User('shared@foo.bar')
+        self.assertEqual(a.shared_with, [u])
+
+        # deselecting
+        args = {'deselected': u.id, 'id': a.id}
+        response = self.get('/analysis/sharing/', args)
+        self.assertEqual(response.code, 200)
+        exp = {'users': [], 'links': ''}
```

Comment:
I think that's a fair point and perhaps worth it. Note that this will imply that we will need to have 2 calls for the same functionality as right now the call actually also changes the display so the new code will have to be: delete/add user, request current values. 

Now, based on (1) this code/mechanism is based on Artifact sharing, which means that we will need to change both to be complete and (2) that what I did is simply return these deleted lines from this branch + add some tests; I  also believe that this is out of the scope of this PR and opening an issue for further discussion is the best path forward.  See https://github.com/biocore/qiita/issues/2171.